src/gentropy/dataset/dataset.py

```diff
@@ -8,7 +8,9 @@
 from functools import reduce
 from typing import TYPE_CHECKING, Any
 
-import pyspark.sql.functions as f
+from pyspark.sql import DataFrame
+from pyspark.sql import functions as f
+from pyspark.sql import types as t
```

Comment:
Extremely marginal, but you are importing all types as t, while in line 14 there's an explicit import of DoubleType. It can be made consistent.